File: C/easy/length_of_last_word.c
int lengthOfLastWord(char * str){

    if (!str) return 0;
    if (str[0] == '\0') return 0;
    
    int i = 0, last_non_sp = 0;
    int len = 0;
  
    while (str[i] != '\0')
    {
        if (str[i] != ' ')
        {
            if (i && str[i-1] == ' ')
                last_non_sp = i;
            else if(i == 0 && str[i] != ' ')
                last_non_sp = i;
        }
        i++;
    }

    i--;
    while (i >= 0)
    {
        if (str[i] == ' ') i--;
        else break;
    }

    return i - last_non_sp + 1;
    
}

File: Python/easy/length_of_last_word.py
# Method 1
class Solution(object):
    def lengthOfLastWord(self, s):
        """
        :type s: str
        :rtype: int
        """
        if len(s.strip().split(" ")) >= 1:
            return len(s.strip().split(" ")[len(s.strip().split(" ")) - 1])
        return 0


# Method 2
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        
        lastInd = len(s) - 1
        
        whiteSpaces = 0                                   ##Pointer that counts the white spaces. 
        while lastInd > 0 and s[ lastInd ] == ' ':
            lastInd -= 1
            whiteSpaces += 1
        
        while lastInd >= 0 and s[ lastInd ] != ' ':       ##Pointer that counts until the end or til the next white space 
            lastInd -= 1
            
        return len(s) - lastInd - whiteSpaces - 1
